### PR TITLE
🚀 Add event values to events in docs

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

- Added event values to events in docs
- Added `EventValueInfo#getTime()`
- Added `EventValueInfo#isEventExcluded(Class)`
- Overloaded `EventValueInfo` constructor and deprecated the old one due to `time` param
- Added `EventValues#getPerEventEventValues()` for getting all events values per event -- useful for docs

_Preview_
image
_nvm about the broken if condition, that's from my side_

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:** 
- closes https://github.com/SkriptLang/skript-docs/issues/9

### CODE REVIEW

File: src/main/java/ch/njol/skript/registrations/EventValues.java [hunk 1]
```diff
@@ -355,5 +394,23 @@ public T get(E e) {
 	public static boolean doesEventValueHaveTimeStates(Class<? extends Event> e, Class<?> c) {
 		return getEventValueGetter(e, c, -1, false) != null || getEventValueGetter(e, c, 1, false) != null;
 	}
+
+	/**
+	 * @return All the event values for each registered event.
+	 */
+	public static Map<Class<? extends Event>, List<EventValueInfo<?, ?>>> getPerEventEventValues() {
+		Map<Class<? extends Event>, List<EventValues.EventValueInfo<?, ?>>> eventValuesCache = new HashMap<>();
+		for (int i = -1; i < 2; i++) {
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/registrations/EventValues.java:403]:
```suggestion
		for (int i = TIME_PAST; i <= TIME_FUTURE; i++) {
```
maybe? :)

File: src/main/java/ch/njol/skript/doc/HTMLGenerator.java [hunk 2]
```diff
@@ -603,6 +617,31 @@ private String generateEvent(String descTemp, SkriptEventInfo<?> info, @Nullable
...
+					continue;
+				for (EventValues.EventValueInfo<?, ?> eventValueInfo : eventValuesCache.get(index)) {
+					if (eventValueInfo.isEventExcluded(event))
+						continue;
+					String prefix = "";
+					if (eventValueInfo.getTime() == EventValues.TIME_PAST)
+						prefix = "past ";
+					else if (eventValueInfo.getTime() == EventValues.TIME_FUTURE)
+						prefix = "future ";
+					String className = eventValueInfo.c.getSimpleName().toLowerCase(Locale.ENGLISH);
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/doc/HTMLGenerator.java:634]: should this use the ClassInfo name instead of replacing?

**Review by APickledWalrus — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/doc/HTMLGenerator.java [hunk 3]
```diff
@@ -603,6 +617,31 @@ private String generateEvent(String descTemp, SkriptEventInfo<?> info, @Nullable
...
+					eventValuesSet.add(prefix + "event-" + (className.contains("[]") ? Utils.toEnglishPlural((className.replace("[]", ""))) : className));
+				}
+			}
+		}
+		List<String> eventValues = new ArrayList<>(0);
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/doc/HTMLGenerator.java:639]:
```suggestion
		List<String> eventValues = new ArrayList<>();
```
No need to initialize the array list here, unless you want to do eventValuesSet.size()

File: src/main/java/ch/njol/skript/registrations/EventValues.java [hunk 4]
```diff
@@ -89,13 +106,35 @@ public Class<? extends E>[] getExcludes() {
...
+		/**
+		 * Whether the provided event is one of the excluded events of this {@link EventValueInfo}
+		 * @return Either {@link #TIME_PAST}, {@link #TIME_NOW} or {@link #TIME_FUTURE}
+		 */
+		public boolean isEventExcluded(Class<? extends Event> c) {
+			if (excludes != null)
+				for (Class<? extends Event> exclude : excludes) {
+					if (exclude.isAssignableFrom(c))
+						return true;
+				}
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/registrations/EventValues.java:134]: Java won't run the for loop if null
```suggestion
			for (Class<? extends Event> exclude : excludes) {
				if (exclude.isAssignableFrom(c))
					return true;
			}
```

File: src/main/java/ch/njol/skript/doc/HTMLGenerator.java [hunk 5]
```diff
@@ -68,6 +75,7 @@ public class HTMLGenerator {
 	private final File template;
 	private final File output;
 	private final String skeleton;
+	private Map<Class<? extends Event>, List<EventValues.EventValueInfo<?, ?>>> eventValuesCache = new HashMap<>();
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/doc/HTMLGenerator.java:78]: Should clear this cache after the templates have been generated, we don't need it sitting in here.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]

**Comment by sovdeeth** [comment]
closing due to inactivity

**Pull request closed without merging**